studio/components/grid/components/header/RefreshButton.tsx

```diff
@@ -1,20 +1,22 @@
-import { FC, useState, useEffect } from 'react'
-import { Button, IconRefreshCw, IconCheck } from 'ui'
+import { useQueryClient, useIsFetching } from '@tanstack/react-query'
+import { SupaTable } from 'components/grid/types'
+import { sqlKeys } from 'data/sql/keys'
+import { useEffect, useState } from 'react'
+import { Button, IconCheck, IconRefreshCw } from 'ui'
 import { SupabaseGridQueue } from '../../constants'
-import { Filter, Sort } from 'components/grid/types'
-import { useDispatch, useTrackedState } from 'components/grid/store'
-import { fetchCount, fetchPage } from 'components/grid/utils'
 
-interface Props {
-  sorts: Sort[]
-  filters: Filter[]
+export type RefreshButtonProps = {
+  projectRef?: string
+  table: SupaTable
 }
 
-const RefreshButton: FC<Props> = ({ sorts, filters }) => {
-  const state = useTrackedState()
-  const dispatch = useDispatch()
+const RefreshButton = ({ projectRef, table }: RefreshButtonProps) => {
```

Comment:
nit: possible to use {ref} from useParams()?
just to keep props to the minimal

Reply:
I did it this way because I was trying to make the component more "reusable". Since if we grab the `projectRef` from `useParams()`, this component _must_ be used on a page that has a `ref` component in the URL.

As for this particular case, I think it's kinda redundant as where else are we going to use this component? So I'm very happy to make this change if you think it's cleaner?

Reply:
ahh okay ^ yeah let's just get the ref from useParams then in this case